### ISSUE TITLE
How can I use dotnet ef dbcontext scaffold

### ISSUE THREAD

**Issue by gaoconggit** [BlueshiftSoftware/EntityFrameworkCore#28]
How can I use dotnet ef dbcontext scaffold？

**Comment by crhairr** [comment]
### tl;dr
You can't.

### Explanation
`dotnet ef` was designed by the original EF Core team to work exclusively with the SQL database providers. Its design has *never* been intended even for extension by non-SQL providers. The concept behind it also will not work very well with a document database.

The most important thing to remember is that MongoDB does not have any concept of schema. So in order to determine a schema from the database, you'd have to scan every single document and accumulate all of the available properties. For even moderately sized databases, such an operation would be prohibitively long.

I have batted around some ideas for how to implement something similar for EFCore-MongoDB. However, I keep coming up short. I'm open to suggestions.
